### PR TITLE
[GPU] bugfix: block copy accessed out of valid position.

### PR DESCRIPTION
**Pull request by sungeunk**
### Tickets:
 - 97947

### CODE REVIEW

**Comment by sungeunk** [comment]
No perf drop.

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/resample_opt.cl [hunk 1]
```diff
@@ -361,8 +363,15 @@ KERNEL (resample_opt)(__global INPUT0_TYPE* input,
 
 #if OUTPUT_DIMS == 5
         WRITE_FUNC(output, OUTPUT_GET_INDEX(b, feature_block, z, y, (x + out_x)), out);
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/resample_opt.cl:365]: Please handle 5d case, too

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/resample_opt.cl [hunk 2]
```diff
@@ -361,8 +363,15 @@ KERNEL (resample_opt)(__global INPUT0_TYPE* input,
...
+#else
+#if (OUTPUT_FEATURE_NUM % FEATURE_SLICE_SIZE) != 0
+        if (FEATURE_LIMIT > feature_num)
+            WRITE_FUNC(output, OUTPUT_GET_INDEX(b, feature_block, y, (x + out_x)), out);
+        else
+            output[OUTPUT_GET_INDEX(b, feature_num, y, (x + out_x))] = out;
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/resample_opt.cl:371]: Please add unittest for this

Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/resample_opt.cl:371]: Let's check performance impact from unittest.

**Review by isanghao — CHANGES_REQUESTED** [review]

**Review by isanghao — APPROVED** [review]

**Pull request merged**